src/core_plugins/tile_map/public/geohash_layer.js

```diff
@@ -40,10 +40,16 @@ export class GeohashLayer extends KibanaMapLayer {
         this._geohashMarkers = new GeohashGridMarkers(this._geohashGeoJson, markerOptions, this._zoom, this._kibanaMap);
         break;
       case 'Heatmap':
+        let radius = 25;
+        if (this._geohashGeoJson.properties.geohashGridDimensionsAtEquator) {
+          const minGridLength = _.min(this._geohashGeoJson.properties.geohashGridDimensionsAtEquator);
+          const metersPerPixel = this._kibanaMap.getMetersPerPixel();
+          radius = minGridLength / metersPerPixel;  // more radius extends into neighboring grid(s) allows for more clustering
+        }
         this._geohashMarkers = new HeatmapMarkers(this._geohashGeoJson, {
-          radius: parseFloat(this._geohashOptions.heatmap.heatRadius),
-          blur: parseFloat(this._geohashOptions.heatmap.heatBlur),
-          maxZoom: parseFloat(this._geohashOptions.heatmap.heatMaxZoom),
+          radius: radius,
+          blur: radius * 0.75,
+          maxZoom: this._kibanaMap.getZoomLevel(),
```

Comment:
I think we should not use the zoom-level of the current map (because that may change for a given precision).  Instead, the lowest zoom level that correspond to the current ES-precision on the map. I think that would yield more consistent results. It would make the construction for new heatmap-marker layer consistent for every ES-precision-result . By using the current zoom level, depending on where you are in the zoom,  this layer could get instantiated with any of the 2-3 zoom levels that correspond to a given ES-precision.